### PR TITLE
Bump OpenSSL to 1.0.1k

### PR DESCRIPTION
**Pull request by alex**
Fixes https://www.openssl.org/news/secadv_20150108.txt

### CODE REVIEW

**Comment by alex** [comment]
I'm not sure I understand the test failures here -- it looks like maybe the lastpass-cli is unhappy?

File: Library/Formula/openssl.rb [hunk 1]
```diff
@@ -1,8 +1,7 @@
 class Openssl < Formula
   homepage "https://openssl.org"
-  url "https://www.openssl.org/source/openssl-1.0.1j.tar.gz"
-  mirror "https://raw.githubusercontent.com/DomT4/LibreMirror/master/OpenSSL/openssl-1.0.1j.tar.gz"
-  sha256 "1b60ca8789ba6f03e8ef20da2293b8dc131c39d83814e775069f02d26354edf3"
+  url "https://www.openssl.org/source/openssl-1.0.1k.tar.gz"
+  sha256 "8f9faeaebad088e772f4ef5e38252d472be4d878c6b3a2718c10a4fcebe7a41c"
   revision 1
```
Comment (tdsmith) [inline Library/Formula/openssl.rb]: nuke this please

Comment (alex) [inline Library/Formula/openssl.rb]: done

**Comment by alex** [comment]
So yeah, it looks like there's something wrong with LastPass, a file does not exist error, I don't really understand how it could have been caused by this change.

**Comment by DomT4** [comment]
I've updated my mirror, so that line can go back in if you like, or we can use another mirror.

Please revision ` tor ` and ` freeradius-server ` as well in this PR. They tantrum horribly and loudly at runtime if the OpenSSL they were compiled against is updated or downgraded.

I wonder if this:

```
Testing lastpass-cli
==> /usr/local/Cellar/lastpass-cli/0.4.0/bin/lpass --version

/usr/local/Cellar/lastpass-cli/0.4.0/bin/lpass
--version

Error: No such file or directory: mkdir(/private/tmp/john-jumbo-QNYSYd/.lpass)
Error: lastpass-cli: failed
Failed executing: /usr/local/Cellar/lastpass-cli/0.4.0/bin/lpass --version
```

happens because ` john-jumbo ` defines a ` HOME ` env, and perhaps that's persisting through tests accidentally? @mikemcquaid

**Comment by alex** [comment]
@DomT4 "https://raw.githubusercontent.com/DomT4/LibreMirror/master/OpenSSL/openssl-1.0.1k.tar.gz" is a 404 for me, am I getting the URL wrong?

Going to bump tor and freeradius now.

**Comment by DomT4** [comment]
@alex Should be up now. Github takes a moment between me updating the mirror and the ` raw ` subdomain realizing there's a new link there.

Cheers for the revisions!

**Comment by alex** [comment]
@DomT4 cheers, thanks.

**Comment by tdsmith** [comment]
How do you provoke the tor warning message? Installing tor, running tor, upgrading openssl, and running tor again, I don't see any new warnings.

**Comment by DomT4** [comment]
Hey, neat, It's gone. I didn't see anything about this changing in their changelog but perhaps they stealth fixed it. You can see the old error in #33197.

**Comment by alex** [comment]
Does that mean I remove the `revision 1` for `tor`? Will everything handle it correctly?

**Comment by tdsmith** [comment]
Maybe don't worry about it; I'll just delete the commit when I pull.

**Comment by alex** [comment]
Fancy!

**Comment by alex** [comment]
I have no idea what to do about the `lastpass-cli` issue.

**Comment by mikemcquaid** [comment]
Can you link me to the problem you're seeing with it?

**Comment by mikemcquaid** [comment]
Never mind, found it. You can ignore it, it's a test that is failing.

**Comment by mikemcquaid** [comment]
Although it's not failing locally so perhaps it needs revisioned too.

**Comment by alex** [comment]
@mikemcquaid I'll take a pass at revisioning it.

**Comment by tdsmith** [comment]
I think Dominyk nailed the cause of the failure above: https://github.com/Homebrew/homebrew/pull/35665#issuecomment-69216647

**Comment by mikemcquaid** [comment]
> happens because john-jumbo defines a HOME env, and perhaps that's persisting through tests accidentally? @mikemcquaid

Yep, guess so. That'll need tweaked, thanks.

**Comment by mikemcquaid** [comment]
Missed that, thanks.

**Comment by alex** [comment]
Is that something you want me to do, or should it be in a separate PR?

**Comment by mikemcquaid** [comment]
I'll do it, don't worry, thanks.

**Comment by alex** [comment]
Thanks!

**Comment by tdsmith** [comment]
Okay to move this as-is, @mikemcquaid?

**Comment by mikemcquaid** [comment]
@tdsmith Needs rebuilt as bottles weren't built for Yosemite/Mountain Lion. On it.

**Comment by tdsmith** [comment]
Just caught that; thanks. Over and out. :)

**Comment by alex** [comment]
Cheers, thanks!

**Pull request closed without merging**